lxl-web/src/lib/components/Matomo.svelte

```diff
@@ -0,0 +1,51 @@
+<script lang="ts">
+	import { env } from '$env/dynamic/public';
+	import { afterNavigate } from '$app/navigation';
+	import { setContext } from 'svelte';
+	import { writable } from 'svelte/store';
+	import { type MatomoTracker } from '$lib/types/matomo';
+
+	const URL: string = env.PUBLIC_MATOMO_URL;
+	const MATOMO_ID: number = +env.PUBLIC_MATOMO_ID;
+
+	const tracker = writable<MatomoTracker>();
+	setContext('matomo', tracker);
+
+	async function initMatomo() {
+		const matomo = window.Matomo;
+
+		if (matomo && MATOMO_ID) {
+			const track = matomo.getTracker(`${URL}/matomo.php`, MATOMO_ID);
+
+			if (track) {
+				track.disableCookies(); // TODO - remove when cookie consent implemented
+				track.enableLinkTracking();
+
+				// add to matomo context store
+				tracker.set(track);
+				track.trackPageView();
+			}
+		}
+	}
+
+	afterNavigate(async ({ to, from, type }) => {
+		if (type === 'enter' || !$tracker) {
+			await initMatomo();
+		} else {
+			if (from?.url.href) {
+				$tracker.setReferrerUrl(from.url.href);
+			}
+			if (to?.url.href) {
+				$tracker.setCustomUrl(to.url.href);
+			}
```

Comment:
Is it a problem that these lines won't be called for the initial navigation (where `initMatamo` is called instead?) Should they also be called after the initialization of the tracker?

Reply:
I would't say is a problem. In the case of `type === 'enter'` (i.e initial page load) there is no `from.url` so we can't set that manually. (I hope Matomo tracks referrals from external pages automatically, but let's see)

In the case of `setCustomUrl`, it's a way to make sure it's the new url that gets tracked, not the old. In the case of `enter` (`initMatomo`), this is not a problem, it detects the current url correctly.